sdk/core/core-rest-pipeline/src/retryStrategies/throttlingRetryStrategy.ts

```diff
@@ -28,28 +34,93 @@ function parseRetryAfterHeader(headerValue: string): number | undefined {
   }
 }
 
+/**
+ * Returns the number of milliseconds to wait based on "retry-after-ms" or "x-ms-retry-after-ms" header value.
+ * Returns undefined if there is no valid value.
+ * @internal
+ */
+function parseRetryAfterMSHeader(retryAfterMSHeaderValue: string | undefined): number | undefined {
+  const delayValueString = retryAfterMSHeaderValue;
+
+  if (delayValueString == null) {
+    return;
+  }
+
+  const delayValueMs: number = Number(delayValueString);
+
+  if (Number.isNaN(delayValueMs)) {
+    return;
+  }
+
+  return delayValueMs;
+}
+
+/**
+ * Type denoting whether a reponse is throttling retry response along with the `retryAfterInMs` value if it is indeed a throttling retry response.
+ * @internal
+ */
+type ThrottlingInfoInternal =
+  | {
+      isThrottlingRetryResponse: false;
+    }
+  | { isThrottlingRetryResponse: true; retryAfterInMs: number };
+
+/**
+ * A response is a throttling retry response if it has a throttling status code (429 or 503),
+ * as long as one of the [ "Retry-After" or "retry-after-ms" or "x-ms-retry-after-ms" ] headers has a valid value.
+ *
+ * Returns an object denoting whether a reponse is throttling retry response along with the `retryAfterInMs` value if it is indeed a throttling retry response.
+ *
+ * @internal
+ */
+function getThrottlingInfo(response?: PipelineResponse): ThrottlingInfoInternal {
+  const defaultInfo: ThrottlingInfoInternal = {
+    isThrottlingRetryResponse: false,
+  };
+
+  if (!(response && [429, 503].includes(response.status))) return defaultInfo;
+
+  const retryAfterHeader = response.headers.get("Retry-After"); // in seconds
+
+  if (retryAfterHeader) {
+    const parsedRetryAfterHeader = parseRetryAfterHeader(retryAfterHeader);
+    if (parsedRetryAfterHeader) {
+      return { isThrottlingRetryResponse: true, retryAfterInMs: parsedRetryAfterHeader };
+    }
+  }
+
+  const retryAfterMSHeader =
+    response.headers.get(RetryAfterMillisecondsHeaders[0]) ?? // in milliseconds
+    response.headers.get(RetryAfterMillisecondsHeaders[1]); // in milliseconds
```

Comment:
nit: `RetryAfterMillisecondsHeaders` is defined as array, perhaps it can be looped through (in a little helper function perhaps) instead of being accessed with hardcoded indices?

Reply:
And that way you can have that single helper also try the default retry header eliminating another branch 😄

Reply:
Updated as suggested, please review again.